v3/testdata/push_down

```diff
@@ -0,0 +1,39 @@
+exec
+CREATE TABLE a (x INT, y INT)
+
+exec
+CREATE TABLE b (x INT, z INT)
+
+push_down
+SELECT a.y, b.z FROM a JOIN b ON (a.x = b.x) WHERE a.x > 7 AND b.z = 3
+----
+projectOp [in=1,3 out=1,3 cols=1,3]
+  project:
+    variable (a.y) [in=1 out=1]
+    variable (b.z) [in=3 out=3]
+  inputs:
+    selectOp [in=0-3 out=0-3 cols=0,1,2,3]
+      inputs:
+        inner join [in=0-3 out=0-3 cols=0,1,2,3]
+          filter:
+            comp (=) [in=0,2]
+              inputs:
+                variable (a.x) [in=0 out=0]
+                variable (b.x) [in=2 out=2]
+          inputs:
+            scan (a (x, y)) [in=0-1 out=0-1 cols=0,1]
+              filter:
```

Comment:
Note that push_down still creates filters on scan nodes. I could get rid of this, but it would require a bit of reworking of `pushDownFilters`. Probably worth doing because as it stands, we could create filters on any relational node (e.g. `projectOp`), which seems bad.

Reply:
Short story: I'd rework it and use selectOp and projectOp.

Long story: The flexibility is confusing here but It may prove useful when a plan is selected from the Memo and we hand it over to code generation.  Such final step in optimization likes grabbing more than one expr node when deciding how to assemble a single plan node (e.g., grabbing a selection and "pegging" it as a residual predicate of a node that is itself selective, such as a join).